mitodl/mitxonline
# Add courses api filters to catalog page

# What are the relevant tickets?




https://github.com/mitodl/hq/issues/2329

# Description (What does it do?)


Adds additional `page__live` and `courserun_is_enrollable` filters to `/api/courses` to satisfy the course display logic in https://github.com/mitodl/hq/issues/1676#issuecomment-1655795103
- `page__live` covers 1  for Course About page 
     - this is default filter name, but can be changed if needed
-  `courserun_is_enrollable` covers 2-6 to narrow down the `course runs` that meet the following criteria
      - run is live
      - start date is not null
      - enrollment_start is in the pass
      - enrollment_end is either null or in the future

Updates the catalog page to call `page=${page}&live=true&page__live=true&courserun_is_enrollable=true`. It doesn't remove the JS logics as I am not sure if we want to do it in this PR, but API call should give the course result we need for frontend, the JS logic will not filter anything

Adds index to courserun.live 

# How can this be tested?

Adds more than 12 courses, make fewer than 12 of them meet the criteria indicated in https://github.com/mitodl/hq/issues/1676#issuecomment-1655795103
Go to http://mitxonline.odl.local:8013/catalog/, notice that page should display these courses that meet the criterial when initially loaded
